vendor/lib/src/config.dart

```diff
@@ -0,0 +1,349 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert' show json;
+import 'package:pub_semver/pub_semver.dart' show Version;
+import 'package:meta/meta.dart' show sealed;
+import 'package:json_annotation/json_annotation.dart';
+import 'package:checked_yaml/checked_yaml.dart';
+import 'package:yaml_edit/yaml_edit.dart' show YamlEditor, wrapAsYamlNode;
+import 'package:yaml/yaml.dart' show CollectionStyle;
+import 'package:vendor/src/version.dart' show packageVersion;
+
+part 'config.g.dart';
+
+const _defaultIncludePatterns = {
+  'pubspec.yaml',
+  'README.md',
+  'LICENSE',
+  'CHANGELOG.md',
+  'lib/**',
+  'analysis_options.yaml',
+};
+
+/// State of the `lib/src/third_party/` folder, stored in
+/// `lib/src/third_party/vendor-state.yaml`.
```

Comment:
Yeah, this could really go both ways, I think I'm repeating the mistake that `pubspec.lock` made, because I kind of want to be consistent.

But really, it's just so much nicer to read in YAML :D